# Домашнее задание 4 по теме:
# «Предпосылки Гаусса-Маркова. Проверка нормальности случайных отклонений»

# Выполнил:
# Радайкин Артём, ПМ20-4 

## Загружаем данные

In [ ]:
install.packages('stats')
library('stats')

install.packages('tseries')
library('tseries')

install.packages('nortest')
library('nortest')

library('lmtest')

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Warning message:
“package ‘stats’ is a base package, and should not be updated”
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
df = read.table('Данные_дз_4.txt', header=TRUE, sep='\t', dec=',', encoding='UTF-8')
head(df,n=5)

Year X   Y        
1 2000 4.0 4.0771595
2 2001 4.7 0.9543387
3 2002 5.8 1.6959429
4 2003 6.0 2.7962091
5 2004 5.5 3.8525526

# Задания:

In [ ]:
# Тест Шапиро-Уилка
# library(stats)
# shapiro.test

# Тест Харке-Бера
# library(tseries)
# jarque.bera.test

# Критерий согласия Пирсона
# library(nortest)
# pearson.test

In [ ]:
model = lm(Y ~ X, data=df)

In [ ]:
summary(model)


Call:
lm(formula = Y ~ X, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.6362 -0.3737  0.4129  0.8816  1.9650 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)   3.8658     1.3928   2.776   0.0141 *
X            -0.3252     0.2277  -1.428   0.1737  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.63 on 15 degrees of freedom
Multiple R-squared:  0.1197,	Adjusted R-squared:  0.06101 
F-statistic:  2.04 on 1 and 15 DF,  p-value: 0.1737


In [ ]:
yi_hat = predict(model)
vector_Se = yi_hat - df[,3]

# Тест Шапиро-Уилка

In [ ]:
shapiro.test(vector_Se)


	Shapiro-Wilk normality test

data:  vector_Se
W = 0.87113, p-value = 0.0229


# Тест Харке-Бера

In [ ]:
jarque.bera.test(vector_Se)


	Jarque Bera Test

data:  vector_Se
X-squared = 10.423, df = 2, p-value = 0.005454


# Критерий согласия Пирсона

In [ ]:
pearson.test(vector_Se)


	Pearson chi-square normality test

data:  vector_Se
P = 7.2941, p-value = 0.1211


## **Выводы:**
Т.к. наши данные содержат выборку только за 17 лет о уровне безработицы и о росте ВВП => размер выборки слишком мал (<50) и при примении критерия Пирсона (X^2) эмпирические частоты могут оказаться заниженными, в результате чего 
будет получен некорректный результат. Это подверждают и наши результаты, только у критерия Пирсона p-value = 0.1211 > 0.05 и соответсвенно гипотеза H0 о том, что вектор случайных отклонений модели имеет нормальное распределение -  принимается. При использовании двух других тестов p-value < 0.05 и гипотеза отвергается. 

Тест Шапиро-Уилка будет лучше всего подходить для нашей задачи, т.к. данные измерены в количественной шкале, а также объём выборки небольшой, < 50.
По результатам этого теста p-value = 0.0229 < 0.05 и гипотеза H0 о том, что вектор случайных отклонений модели имеет нормальное распределение - отвергается. Это и будет мой окончательный вывод по данной задаче. 